pilot/pkg/serviceregistry/kube/controller/util.go

```diff
@@ -189,3 +190,13 @@ func isNodePortGatewayService(svc *v1.Service) bool {
 	_, ok := svc.Annotations[kube.NodeSelectorAnnotation]
 	return ok && svc.Spec.Type == v1.ServiceTypeNodePort
 }
+
+// Get the proxy pod key which can be used to get pod from the informer cache
+func proxyPodKey(proxy *model.Proxy) string {
+	parts := strings.Split(proxy.ID, ".")
```

Comment:
This is leaking implementation details IMO. In the metadata parsing code we even say getting info from proxy.ID is legacy and we push all info into NodeMetadata. At the very least, we should not be doing strings.Split here and instead should have some sort of field in model.Proxy so we don't have to parse it here every time

Reply:
Also I am worried this is backwards incompatible, I know there are many tools I have written that will be broken by this change

Reply:
Agree for some customized proxy, they may be formatted, so i leave the legacy getPodByIP and if it fails here, it does not matter.